app/models/mvi.rb

```diff
@@ -68,16 +69,16 @@ def select_source_id(correlation_id)
   end
 
   def mvi_response
-    return nil unless @user.loa3? && !user.gender.nil?
-    response || query_and_cache_response
+    return nil unless @user.loa3?
+    @memoized_response ||= response || query_and_cache_response
   end
 
   def mvi_service
     @service ||= MVI::ServiceFactory.get_service(mock_service: ENV['MOCK_MVI_SERVICE'])
   end
 
   def create_message
-    raise Common::Exceptions::ValidationErrors, @user unless @user.valid?
+    raise Common::Exceptions::ValidationErrors, @user unless @user.valid?(:loa3?)
```

Comment:
Is this supposed to perform the LOA3 validation? I don't think it is doing so:
```
2.3.0 :001 > u = User.new(uuid: "123456", email: "bill@aol.com", loa: {highest: 1, current: 1})
 => #<User:0x000000023b1fd0 @uuid="123456", @email="bill@aol.com", @loa={:highest=>1, :current=>1}, @first_name=nil, @middle_name=nil, @last_name=nil, @gender=nil, @birth_date=nil, @zip=nil, @ssn=nil, @last_signed_in=nil, @mhv_last_signed_in=nil, @persisted=false> 
2.3.0 :002 > u.valid?
 => true 
2.3.0 :003 > u.valid?(:loa3)
 => true 
2.3.0 :004 >
```

Coincidentally, I'm doing something very similar but I had to change this inside User.rb from:
```ruby
with_options unless: :loa1? do |user|
```
to this,
```ruby
with_options(on: :loa3_user) do |user|
```
and then I could call:
```ruby
@user.valid?             # perform only LOA 1 valiation
@user.valid?(:loa3_user) # perform LOA 1 & LOA 3 validation
```